app/views/entities/show.rb

```diff
@@ -42,26 +42,30 @@ def include_nav_searchbar
     def head_meta
       mustache[:head_meta] ||= begin
         title = page_title
-        description = truncate(entity_description, length: 350, separator: ' ')
-        description = description.strip! || description
-        thumbnail = entity_thumbnail
-        image = thumbnail ? (thumbnail[:src] || thumbnail[:full] || nil) : nil
-        url = entities_fetch_url(@entity_params)
         head_meta = [
-          { meta_name: 'description', content: description },
+          { meta_name: 'description', content: head_meta_description },
           { meta_property: 'fb:appid', content: '185778248173748' },
           { meta_name: 'twitter:card', content: 'summary' },
           { meta_name: 'twitter:site', content: '@EuropeanaEU' },
           { meta_property: 'og:sitename', content: title },
-          { meta_property: 'og:description', content: description },
-          { meta_property: 'og:url', content: url }
+          { meta_property: 'og:description', content: head_meta_description },
+          { meta_property: 'og:url', content: request.original_url }
         ]
         head_meta << { meta_property: 'og:title', content: title } unless title.nil?
-        head_meta << { meta_property: 'og:image', content: image } unless image.nil?
+        head_meta << { meta_property: 'og:image', content: og_image } unless og_image.nil?
         head_meta + super
       end
     end
 
+    def head_meta_description
+      truncate(entity_description, length: 350, separator: ' ')
```

Comment:
was there no need for the ```.strip``` in the end?
Also doing this will stop running the truncation twice:
```
mustache[:head_meta_description] =|| begin
  truncate(entity_description, length: 350, separator: ' ')
end
```